src/Mod/Sketcher/Gui/TaskSketcherMessages.cpp

```diff
@@ -41,152 +42,113 @@ using namespace Gui::TaskView;
                "partially redundant constraints."));
+    }
 }
 
-void TaskSketcherMessages::onLabelConstrainStatusLinkClicked(const QString& str)
+void TaskSketcherMessages::onLabelStatusLinkClicked(const QString& str)
 {
-    if (str == QStringLiteral("#conflicting"))
-        Gui::Application::Instance->commandManager().runCommandByName(
-            "Sketcher_SelectConflictingConstraints");
-    else if (str == QStringLiteral("#redundant"))
-        Gui::Application::Instance->commandManager().runCommandByName(
-            "Sketcher_SelectRedundantConstraints");
-    else if (str == QStringLiteral("#dofs"))
-        Gui::Application::Instance->commandManager().runCommandByName(
-            "Sketcher_SelectElementsWithDoFs");
-    else if (str == QStringLiteral("#malformed"))
-        Gui::Application::Instance->commandManager().runCommandByName(
-            "Sketcher_SelectMalformedConstraints");
-    else if (str == QStringLiteral("#partiallyredundant"))
-        Gui::Application::Instance->commandManager().runCommandByName(
-            "Sketcher_SelectPartiallyRedundantConstraints");
+    if (str == QStringLiteral("#conflicting")) {
+        Gui::Application::Instance->commandManager().runCommandByName("Sketcher_SelectConflictingConstraints");
+    }
+    else if (str == QStringLiteral("#redundant")) {
+        Gui::Application::Instance->commandManager().runCommandByName("Sketcher_SelectRedundantConstraints");
+    }
+    else if (str == QStringLiteral("#dofs")) {
+        Gui::Application::Instance->commandManager().runCommandByName("Sketcher_SelectElementsWithDoFs");
+    }
+    else if (str == QStringLiteral("#malformed")) {
+        Gui::Application::Instance->commandManager().runCommandByName("Sketcher_SelectMalformedConstraints");
+    }
+    else if (str == QStringLiteral("#partiallyredundant")) {
+        Gui::Application::Instance->commandManager().runCommandByName("Sketcher_SelectPartiallyRedundantConstraints");
+    }
 }
 
-void TaskSketcherMessages::onAutoUpdateStateChanged()
+void TaskSketcherMessages::createSettingsButtonActions()
 {
-    bool state = std::as_const(ui->manualUpdate)->actions()[0]->isChecked();
+    QToolButton* btn = getSettingsButton();
+    btn->show();
 
+    // Set Auto Update in the 'Manual Update' button menu.
     ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath(
         "User parameter:BaseApp/Preferences/Mod/Sketcher");
-    hGrp->SetBool("AutoRecompute", state);
+    bool state = hGrp->GetBool("AutoRecompute", false);
+
     sketchView->getSketchObject()->noRecomputes = !state;
-}
 
-void TaskSketcherMessages::onManualUpdateClicked(bool checked)
-{
-    Q_UNUSED(checked);
-    Gui::Command::updateActive();
+    auto* autoUpdateAction = new QWidgetAction(this);
+    auto* containerWidget = new QWidget();
+    auto* layout = new QGridLayout(containerWidget);
+    auto* checkbox = new QCheckBox(tr("Auto Update"));

```

Comment:
Please include `QCheckBox` directly as it can cause compiler errors for some Qt versions